### PR TITLE
Remove JITIL

### PR DESCRIPTION
**Pull request by JosJuice**
* It's unmaintained
* There's essentially no reason for users to use it instead of the regular x86-64 JIT
* It's preventing us from making PIE builds (PR #5271)
* Probably some other reasons too

### CODE REVIEW

File: Source/Core/Core/PowerPC/PowerPC.h [hunk 1]
```diff
@@ -25,10 +25,12 @@ enum CPUCore
 {
   CORE_INTERPRETER,
   CORE_JIT64,
-  CORE_JITIL64,
-  CORE_JITARM,
+  CORE_JITIL64,  // Doesn't exist anymore
```
Comment (lioncash) [inline Source/Core/Core/PowerPC/PowerPC.h]: It'd probably be better to just remove these and explicitly give the enum instances their respective values. i.e.

```cpp
// 2 and 3 used to be used....
// [explanation here]
enum CPUCore
{
  CORE_INTERPRETER = 0,
  CORE_JIT64 = 1,
  CORE_JITARM64 = 4,
  CORE_CACHEDINTERPRETER = 5
};
```

which removes the identifier entirely and prevents it from being accidentally used.

**Comment by waddlesplash** [comment]
Can I get credit for supporting this, even if I was 3 years too early (#1023)? 😛

**Comment by lioncash** [comment]
Reppin' PIE and ASLR city~

**Comment by Kolano** [comment]
Were issues with the PokéPark titles addressed, we claim they need JITIL on the wiki.
https://wiki.dolphin-emu.org/index.php?title=Pok%C3%A9Park_2:_Wonders_Beyond
https://wiki.dolphin-emu.org/index.php?title=Pok%C3%A9Park_Wii:_Pikachu%27s_Adventure

**Pull request merged**